### PR TITLE
[feature] Created Astral Meter Logic + Tests

### LINKED ISSUE #130: Implement Astral Meter Logic
- Exploration Rate (Real world | Astral world)
- 3 | 5 per mnt = 0.05 | 0.083 per s = 25 mnt gameplay
- If someone is in the Astral World, the rate uses the Astral World rate.
- Kill successful = 15-20 (random)
- Kill Phase - In Ghost Line of Sight = 1 per s (perlu dibalance)
- NPC Wrong Answer = 10-15 (random)

### PR DESCRIPTION
**Pull request by VadimBhirawa**
### Main changes
Created Astral Meter Logic Script. Also created Tests for the Script.

### Tasks List
- [close #130]

### CODE REVIEW

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    void PlayerKilled();
+    bool World();
+}
+
+public class AstralMeterLogic : MonoBehaviour, IAstralMeterLogic
+{
+    private float _astralMeter = 0.0f;
+    private float _maxMeter = 100.0f;
+    private float _constantRate = 0.05f;
+    private float sightAmount = 1.0f;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:22]: rename to _sightAmount

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    public bool isOnRealWorld = true;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:23]: set to private, rename to _isOnRealWorld

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    public bool isOnSight = false;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:24]: set to private, rename to _isOnSight

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    // Start is called before the first frame update
+    private void Start()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:27]: delete this

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    {
+
+    }
+
+    public bool World()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:32]: rename to IsOnRealWorld

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            _constantRate = 0.05f;
+        }
+        else
+        {
+            _constantRate = 0.083f;
+        }
+    }
+
+    public bool OnSight()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:60]: rename to IsOnSight

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        return isOnSight;
+    }
+
+    public void ChangeSight()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:65]: rename to ChangeSightState

File: Assets/Astralization/Scripts/Party/AstralMeterLogic.cs [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public void PlayerKilled()
+    {
+        int randKillAmount = Random.Range(15, 20);
+        _astralMeter += (float)randKillAmount;
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Party/AstralMeterLogic.cs:83]: put Update below Start

File: Assets/Astralization/Tests/Interactable Tests/AstralTest.cs [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        yield return new WaitWhile(() => sceneLoaded == false);
+        IAstralMeterLogic astralMeterLogic = party.GetComponent<IAstralMeterLogic>();
+        astralMeterLogic.ChangeWorld();
+        astralMeterLogic.PlayerKilled();
+
+        Assert.IsFalse(astralMeterLogic.World());
+        Assert.IsTrue(astralMeterLogic.GetAstralMeter() >= 15.0f);
+        Assert.IsTrue(astralMeterLogic.GetAstralMeter() <= 20.0f);
+    }
+}
```
Comment (rafiftaris) [inline Assets/Astralization/Tests/Interactable Tests/AstralTest.cs:126]: add new line

**Review by rafiftaris — CHANGES_REQUESTED** [review]

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

**Review by rafiftaris — APPROVED** [review]
TIL jangan pecicilan pas nunggu test

**Pull request merged**